app/Jobs/RemoveBeatmapsetScores.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+declare(strict_types=1);
+
+namespace App\Jobs;
+
+use App\Libraries\Search\ScoreSearch;
+use App\Libraries\Search\ScoreSearchParams;
+use App\Models\Beatmap;
+use App\Models\Beatmapset;
+use App\Models\Score\Best\Model as ScoreBestModel;
+use App\Models\Solo\Score;
+use App\Models\UserStatistics;
+use DB;
+use Ds\Set;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Database\Eloquent\Collection;
+
+class RemoveBeatmapsetScores implements ShouldQueue
+{
+    use Queueable;
+
+    public $timeout = 3600;
+
+    private int $beatmapsetId;
+    private int $maxScoreId;
+    private Set $resetUserRankStats;
+    private array $schemas;
+    private ScoreSearch $scoreSearch;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct(Beatmapset $beatmapset)
+    {
+        $this->beatmapsetId = $beatmapset->getKey();
+        $this->maxScoreId = Score::max('id') ?? 0;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle()
+    {
+        $this->resetUserRankStats = new Set();
+        $this->scoreSearch = new ScoreSearch();
+        $this->schemas = $this->scoreSearch->getActiveSchemas();
+
+        $beatmapIdsQuery = Beatmap::where('beatmapset_id', $this->beatmapsetId)->select('beatmap_id');
+        Score
+            ::with('performance')
+            ->whereIn('beatmap_id', $beatmapIdsQuery)
+            ->where('id', '<=', $this->maxScoreId)
+            ->chunkById(1000, function (Collection $scores): void {
+                foreach ($scores as $score) {
+                    $this->deleteScore($score);
+                }
+            });
+    }
+
+    private function deleteScore(Score $score): void
+    {
+        DB::transaction(function () use ($score): void {
+            $this->resetUserRankStatsFor($score);
+            $score->performance?->delete();
+            $score->delete();
+        });
+        $this->scoreSearch->queueForIndex($this->schemas, [$score->getKey()]);
```

Comment:
Updated... but that caused the scores to be updated to have its preserve flag false first before queuing as otherwise it won't be deleted by indexer process ಠ_ಠ

Reply:
🤔